packages/widget.vite-plugin/src/plugin.ts

```diff
@@ -235,57 +257,52 @@ export function FoundryWidgetVitePlugin(options: Options = {}): Plugin {
-                  accept: "application/json",
-                  "content-type": "application/json",
-                },
-              },
-            )
-              .then((devResponse) => {
-                if (devResponse.status === 200 || devResponse.status === 204) {
-                  res.setHeader("Content-Type", "application/json");
-                  res.end(
-                    JSON.stringify({
-                      redirectUrl:
-                        `${url.origin}/workspace/custom-views/preview/${foundryConfig.widget.rid}`,
-                    }),
-                  );
-                } else {
-                  res.statusCode = devResponse.status;
-                  res.statusMessage =
-                    `Unable to start dev mode in Foundry (see terminal for more): ${devResponse.statusText}`;
-                  devResponse.text().then((err) => {
-                    config?.logger.error(err);
-                    res.end();
-                  });
-                }
-              })
-              .catch((error) => {
-                res.statusCode = 500;
-                res.statusMessage =
-                  `Unable to start dev mode in Foundry: ${error.message}`;
+            const settingsResponse = await setWidgetSettings(
+              // TODO: Actually handle the widget RID from within the config, which will require somehow parsing the config
+              // Unfortunately, moduleParsed is not called during vite's dev mode for performance reasons, so the config file
+              // will need to be parsed/read a different way
+              foundryConfig.widget.rid,
+              foundryUrl,
+              localhostUrl,
+              entrypointToJsSourceFileMap,
+              entrypointFileName,
+            );
+            if (
+              settingsResponse.status !== 200
+            ) {
+              res.statusCode = settingsResponse.status;
+              res.statusMessage =
+                `Unable to set widget settings in Foundry: ${settingsResponse.statusText}`;
+              settingsResponse.text().then((err) => {
+                config?.logger.error(err);
                 res.end();
               });
+              return;
+            }
+
+            const enableResponse = await enableDevMode(foundryUrl);
+            if (enableResponse.status !== 200) {
+              res.statusCode = enableResponse.status;
+              res.statusMessage =
+                `Unable to start dev mode in Foundry: ${enableResponse.statusText}`;
+              res.end();
+              return;
+            }
+
+            try {
+              res.setHeader("Content-Type", "application/json");
+              res.end(
+                JSON.stringify({
+                  redirectUrl:
+                    `${foundryUrl.origin}/workspace/custom-views/preview/${foundryConfig.widget.rid}`,
+                }),
+              );
+            } catch (error: any) {
```

Comment:
What kind of error would this catch? I think in the old code the `.catch` looks like it's targeted on the fetch. Another difference with the `.catch` is it would probably catch if the fetch itself fails to return a response so we may want to add that around the fetch to be the same as before

https://github.com/palantir/osdk-ts/blob/04d5e22299031472d51b1c531a8dbb305aeea3f7/packages/shared.net.fetch/src/createFetchOrThrow.ts#L34-L38

Reply:
Ah yes, the try is in the wrong place